resources/qml/PrintSetupSelector.qml

```diff
@@ -0,0 +1,241 @@
+
+        ButtonGroup
+        {
+            id: modeMenuGroup
+        }
+
+        delegate: Button
+        {
+            id: control
+
+            height: settingsModeSelection.height
+            width: Math.round(parent.width / 2)
+
+            anchors.left: parent.left
+            anchors.leftMargin: model.index * Math.round(settingsModeSelection.width / 2)
+            anchors.verticalCenter: parent.verticalCenter
+
+            ButtonGroup.group: modeMenuGroup
+
+            checkable: true
+            checked: base.currentModeIndex == index
+            onClicked: base.currentModeIndex = index
+
+            onHoveredChanged:
+            {
+                if (hovered)
+                {
+                    tooltipDelayTimer.item = settingsModeSelection
+                    tooltipDelayTimer.text = model.tooltipText
+                    tooltipDelayTimer.start()
+                }
+                else
+                {
+                    tooltipDelayTimer.stop()
+                    base.hideTooltip()
+                }
+            }
+
+            background: Rectangle
+            {
+                border.width: control.checked ? UM.Theme.getSize("default_lining").width * 2 : UM.Theme.getSize("default_lining").width
+                border.color: (control.checked || control.pressed) ? UM.Theme.getColor("action_button_active_border") : control.hovered ? UM.Theme.getColor("action_button_hovered_border") : UM.Theme.getColor("action_button_border")
+
+                // for some reason, QtQuick decided to use the color of the background property as text color for the contentItem, so here it is
+                color: (control.checked || control.pressed) ? UM.Theme.getColor("action_button_active") : control.hovered ? UM.Theme.getColor("action_button_hovered") : UM.Theme.getColor("action_button")
+            }
+
+            contentItem: Label
+            {
+                text: model.text
+                font: UM.Theme.getFont("default")
+                horizontalAlignment: Text.AlignHCenter
+                verticalAlignment: Text.AlignVCenter
+                renderType: Text.NativeRendering
+                elide: Text.ElideRight
+                color:
+                {
+                    if(control.pressed)
+                    {
+                        return UM.Theme.getColor("action_button_active_text");
+                    }
+                    else if(control.hovered)
+                    {
+                        return UM.Theme.getColor("action_button_hovered_text");
+                    }
+                    return UM.Theme.getColor("action_button_text");
+                }
+            }
+        }
+    }
+
+    Item
+    {
+        id: sidebarContents
+        anchors.top: settingsModeSelection.bottom
+        anchors.topMargin: UM.Theme.getSize("sidebar_margin").height
+        anchors.left: parent.left
+        anchors.right: parent.right
+        height: 500
```

Comment:
Make this themeable? The side bar has a size with both a width and a height, right?

Reply:
Done in 5a5adb71cd159c2e7b7204aa40b42472db4d9995. However this is going to change in the next weeks. But at least now is prepared with a setting in the theme.